src/modules/admin/components/Permissions/ColonyPermissionEditDialog.tsx

```diff
@@ -0,0 +1,300 @@
+  },
+  buttonConfirm: {
+    id: 'core.ColonyPermissionEditDialog.buttonConfirm',
+    defaultMessage: 'Confirm',
+  },
+});
+
+interface Props {
+  cancel: () => void;
+  close: () => void;
+  domain: DomainType;
+  /*   selectedUser?: UserType; */
+  colonyAddress: Address;
+  submit: ActionTypeString;
+  success: ActionTypeString;
+  error: ActionTypeString;
+}
+
+const supFilter = (data, filterValue) => {
+  if (!filterValue) {
+    return data;
+  }
+
+  const filtered = data.filter(
+    user =>
+      user &&
+      filterValue &&
+      (user.profile.username
+        .toLowerCase()
+        .includes(filterValue.toLowerCase()) ||
+        user.profile.walletAddress
+          .toLowerCase()
+          .includes(filterValue.toLowerCase())),
+  );
+
+  const customValue = {
+    id: 'filterValue',
+    profile: {
+      walletAddress: filterValue,
+      displayName: filterValue,
+    },
+  };
+
+  return [customValue].concat(filtered);
+};
+
+const validationSchema = yup.object({
+  user: yup.object().required(),
+  domainId: yup.number(),
+  colonyAddress: yup.string().required(),
+  roles: yup.array().of(yup.string().required()),
+});
+
+const ColonyPermissionEditDialog = ({
+  domain,
+  /* selectedUser, */
+  colonyAddress,
+  cancel,
+  close,
+}: Props) => {
+  // Prepare userData for SingleUserPicker
+  const userAddressesInStore = useSelector(allUsersAddressesSelector);
+
+  const userData = useDataMapFetcher<UserType>(
+    usersByAddressFetcher,
+    Array.from(userAddressesInStore),
+  );
+
+  const users = useMemo(
+    () =>
+      userData
+        .filter(({ data }) => !!data)
+        .map(({ data, key }) => ({
+          id: key,
+          ...data,
+        })),
+    [userData],
+  );
+
+  const availableRoles = [
+    'Root',
+    'Administration',
+    'Architecture',
+    'Funding',
+    'Arbitration',
+  ];
+
+  /* Currently arbitration should appear in the list of roles
+    but can not be set yet
+   */
+  const checkIfCanBeSet = role => {
+    if (role === 'Arbitration' || role === 'Root') {
+      return false;
+    }
+    return true;
+  };
+
+  const transform = useCallback(
+    pipe(
+      withKey(colonyAddress),
+      mapPayload(p => ({
+        userAddress: p.user.profile.walletAddress,
+        domainId: p.domainId,
+        roles: p.roles.reduce((accumulator, role) => {
+          if (checkIfCanBeSet(role)) {
+            accumulator[role.toUpperCase()] = true;
+            return accumulator;
+          }
+          return accumulator;
+        }, {}),
+      })),
+      mergePayload({ colonyAddress }),
+    ),
+    [colonyAddress],
+  );
+
+  // When updating the selected user fetch that user's Roles
+  const [currentUser, setCurrentUser] = useState(null);
+
+  const updateCurrentUser = useCallback(({ profile: { walletAddress } }) => {
+    setCurrentUser(walletAddress);
+  }, []);
+
+  // Get current user roles to populate the checkboxes
+  const useCurrentUserForRoles = (): Array<string> => {
```

Comment:
The name of this function uses the convention for hooks, but it's a function. I also find it a little difficult to read, but maybe it can be much simpler with a mapping instead.

Reply:
I kind of believed I made a custom hook here, that's why I called it like that. :-) I must be wrong here. I am still not sure what turns it into a hook actually.